### PR TITLE
[WIP] Adds OAuth proxy and ServiceAccount

### PR DESCRIPTION
**Pull request by josemigallas**
This is basically a replication of what's been done in [metrics-apb].(https://github.com/aerogearcatalog/metrics-apb)

JIRA: https://issues.jboss.org/browse/AEROGEAR-2306

### CODE REVIEW

File: roles/provision-unifiedpush-apb/defaults/main.yml [hunk 1]
```diff
@@ -1,2 +1,9 @@
+# UPS Values
 ups_secret_name: "unified-push-server"
-ups_image_version: "1.3.0"
\ No newline at end of file
+ups_image_version: "1.3.0"
```
Comment (david-martin) [inline roles/provision-unifiedpush-apb/defaults/main.yml]: is this the image with no auth?

Comment (pb82) [inline roles/provision-unifiedpush-apb/defaults/main.yml]: should be 1.3.0-no-auth

File: roles/provision-unifiedpush-apb/tasks/main.yml [hunk 2]
```diff
@@ -131,6 +139,22 @@
           - "-i"
           - "-c"
           - 'wget http://localhost:8080/ag-push '
+        containers:
```
Comment (david-martin) [inline roles/provision-unifiedpush-apb/tasks/main.yml]: I think we're missing the part for telling kubernetes what serviceaccount to run our pod as

e.g. https://github.com/aerogearcatalog/metrics-apb/blob/master/roles/provision-metrics-apb/tasks/provision-prometheus.yml#L36

Comment (david-martin) [reply roles/provision-unifiedpush-apb/tasks/main.yml]: resolved

File: roles/provision-unifiedpush-apb/tasks/main.yml [hunk 3]
```diff
@@ -108,6 +116,7 @@
     spec_template_metadata_labels:
       app: ups
       service: ups
+    spec_template_spec_service_account_name: '{{ proxy_serviceaccount_name }}'
     containers:
     - name: ups
       image: 'docker.io/aerogear/unifiedpush-wildfly:{{ ups_image_version }}'
```
Comment (matzew) [inline roles/provision-unifiedpush-apb/tasks/main.yml]: matzew is the dh org

Comment (matzew) [reply roles/provision-unifiedpush-apb/tasks/main.yml]: See https://github.com/josemigallas/unifiedpush-apb/pull/1

File: roles/provision-unifiedpush-apb/tasks/main.yml [hunk 4]
```diff
@@ -131,6 +140,22 @@
...
+      imagePullPolicy: IfNotPresent
+      ports:
+      - containerPort: '{{ ups_proxy_port }}'
+        name: public
+      args:
+      - --provider=openshift
+      - --openshift-service-account={{ proxy_serviceaccount_name }}
+      - --upstream=http://localhost:{{ ups_port }}
+      - >-
+        --openshift-sar={"namespace":"{{ namespace }}","resource":"deploymentconfigs","name":"ups","verb":"update"}
```
Comment (matzew) [inline roles/provision-unifiedpush-apb/tasks/main.yml:145]: is this for the `update` permission ?

Comment (david-martin) [reply roles/provision-unifiedpush-apb/tasks/main.yml:145]: So this is saying that the oauth proxy will only let users past if they have 'update' permission on the ups deploymentconfig.

**Review by david-martin — APPROVED** [review]
Approving based on it working locally for me.

image

image

image

image


Slight aside, I've created a jira task to generate the mysql passwords rather than have hardcoded defaults https://issues.jboss.org/browse/AEROGEAR-2350

**Pull request merged**